dhis-2/dhis-api/src/main/java/org/hisp/dhis/keyjsonvalue/KeyJsonValueService.java

```diff
@@ -58,72 +89,58 @@
      */
-    KeyJsonValue getKeyJsonValue( String namespace, String key, boolean isAdmin );
+    KeyJsonValue getKeyJsonValue( String namespace, String key );
 
     /**
-     * Adds a new KeyJsonValue.
+     * Adds a new entry.
      *
-     * @param keyJsonValue the KeyJsonValue to be stored.
-     * @return the id of the KeyJsonValue stored.
+     * @param entry the KeyJsonValue to be stored.
+     * @throws IllegalStateException when an entry with same namespace and key
+     *         already exists
+     * @throws IllegalArgumentException when the entry value is not valid JSON
+     * @throws AccessDeniedException when user lacks authority for namespace or
+     *         entry
      */
-    Long addKeyJsonValue( KeyJsonValue keyJsonValue );
+    void addKeyJsonValue( KeyJsonValue entry );
 
     /**
-     * Updates a KeyJsonValue.
+     * Updates an entry.
      *
-     * @param keyJsonValue the updated KeyJsonValue.
+     * @param entry the updated KeyJsonValue.
+     * @throws IllegalArgumentException when the entry value is not valid JSON
+     * @throws AccessDeniedException when user lacks authority for namespace or
+     *         entry
      */
-    void updateKeyJsonValue( KeyJsonValue keyJsonValue );
+    void updateKeyJsonValue( KeyJsonValue entry );
 
     /**
-     * Deletes a keyJsonValue.
+     * Deletes an entry.
      *
-     * @param keyJsonValue the KeyJsonValue to be deleted.
+     * @param entry the KeyJsonValue to be deleted.
+     * @throws AccessDeniedException when user lacks authority for namespace or
+     *         entry
      */
-    void deleteKeyJsonValue( KeyJsonValue keyJsonValue );
+    void deleteKeyJsonValue( KeyJsonValue entry );
 
     /**
-     * Deletes all keys associated with a given namespace.
+     * Deletes all entries associated with a given namespace.
      *
      * @param namespace the namespace to delete
+     * @throws AccessDeniedException when user lacks authority for namespace or
+     *         any of the entries
      */
     void deleteNamespace( String namespace );
 
-    /**
-     * Retrieves a value object.
-     *
-     * @param namespace the namespace where the key is associated.
-     * @param key the key referencing the value.
-     * @param clazz the class of the object to retrievev.
-     * @return a value object.
-     */
-    <T> T getValue( String namespace, String key, Class<T> clazz );
-
-    /**
-     * Adds a value object.
-     *
-     * @param namespace the namespace where the key is associated.
-     * @param key the key referencing the value.
-     * @param value the value object to add.
-     */
-    <T> void addValue( String namespace, String key, T value );
-
-    /**
-     * Updates a value object.
-     *
-     * @param namespace the namespace where the key is associated.
-     * @param key the key referencing the value.
-     * @param value the value object to update.
-     */
-    <T> void updateValue( String namespace, String key, T value );
```

Comment:
These were only used by tests so I moved them to the test as utility. Mostly because they were a bit problematic for the semantics of protection in some cases this seemed like a easy and good way out of it.